pkg/listwatch/listwatch.go

```diff
@@ -0,0 +1,269 @@
+// This was originally copied from the Prometheus Operator project. Due to
```

Comment:
Would there be a license breach if we just add this as 2019 Copyright of Kube-state-metrics? Considering my earlier review comments, we would be modifying the signature of ListWatch method to avoid the `go-kit` dependency and other changes should trickle down. 

Thoughts? cc @LiliC

Reply:
I would keep this, or like suggested above just vendor this in. We want this in ASAP to fix the bug, so I think just merging it ASAP and fixing things in a future release would be best, wdyt @brancz @tariq1890